Lacrit/assignment1/q2.cpp

```diff
@@ -4,44 +4,76 @@
 template<class T> 
 struct Node
 {
-    T i;
-    struct Node *next;
+	private:
+    	T i;
+		Node *next;
+	public: 
+		Node* get_next()
+		{
+			return next;
+		}  
+		T get_i()
+		{
+			return i;
+		}
+		void set_next(Node *new_next)
```

Comment:
The user can still modify next :)
One solution is to make a LinkedList class, and mark is as a friend of Node, that way only there you can modify it.